modules/core/src/scenegraph/scenegraph-node.js

```diff
@@ -40,6 +40,10 @@ export default class ScenegraphNode {
     return this;
   }
 
+  setMatrix(matrixArray) {
+    this.matrix.copy(matrixArray);
```

Comment:
I prefer it that way because the **Matrix** is mutable and can result in hard-to-debug issues.

Reply:
> I prefer it that way because the Matrix is mutable and can result in hard-to-debug issues.

True, though the perf hit can be very significant, and not giving the user the option to do it performantly is not good.

One way out is to add a second argument that controls whether to copy or set, or just add a separate method `copyMatrix`.